homeassistant/components/weather/__init__.py

```diff
@@ -0,0 +1,123 @@
+"""
+Weather component that handles meteorological data for your location.
+
+For more details about this component, please refer to the documentation at
+https://home-assistant.io/components/weather/
+"""
+import logging
+
+from homeassistant.helpers.entity_component import EntityComponent
+from homeassistant.util.temperature import convert as convert_temperature
+from homeassistant.helpers.config_validation import PLATFORM_SCHEMA  # noqa
+from homeassistant.helpers.entity import Entity
+
+_LOGGER = logging.getLogger(__name__)
+
+DEPENDENCIES = []
+DOMAIN = 'weather'
+
+ENTITY_ID_FORMAT = DOMAIN + '.{}'
+
+ATTR_CONDITION_CLASS = 'condition_class'
+ATTR_WEATHER_ATTRIBUTION = 'attribution'
+ATTR_WEATHER_HUMIDITY = 'humidity'
+ATTR_WEATHER_OZONE = 'ozone'
+ATTR_WEATHER_PRESSURE = 'pressure'
+ATTR_WEATHER_TEMPERATURE = 'temperature'
+ATTR_WEATHER_WIND_BEARING = 'wind_bearing'
+ATTR_WEATHER_WIND_SPEED = 'wind_speed'
+
+ATTR_TO_PROPERTY = [
+    ATTR_WEATHER_ATTRIBUTION,
+    ATTR_WEATHER_HUMIDITY,
+    ATTR_WEATHER_OZONE,
+    ATTR_WEATHER_PRESSURE,
+    ATTR_WEATHER_TEMPERATURE,
+    ATTR_WEATHER_WIND_BEARING,
+    ATTR_WEATHER_WIND_SPEED,
+]
+
+
+def setup(hass, config):
+    """Setup the weather component."""
+    component = EntityComponent(_LOGGER, DOMAIN, hass)
+    component.setup(config)
+
+    return True
+
+
+# pylint: disable=no-member, no-self-use, too-many-return-statements
+class WeatherEntity(Entity):
+    """ABC for a weather data."""
+
+    @property
+    def temperature(self):
+        """Return the temperature."""
+        return convert_temperature(self.device_temperature,
+                                   self.temperature_unit,
+                                   self.hass.config.units.temperature_unit)
+
+    @property
+    def device_temperature(self):
```

Comment:
No, `temperature` should be the one that the platform implements. You should convert it in `state_attributes`